tests/test_emulator_codegen.py

```diff
@@ -111,3 +113,269 @@ def test_codegen_detuning_and_rabi():
+
+    geometry = Register(
+        TwoLevelAtom,
+        [(Decimal("0"), Decimal("0")), (Decimal("0"), Decimal("5"))],
+        Decimal("0"),
+    )
+
+    detuning_term_0 = DetuningTerm(
+        DetuningOperatorData({0: Decimal("1")}),
+        compiled_wf_0,
+    )
+
+    detuning_term_1 = DetuningTerm(
+        DetuningOperatorData({1: Decimal("1")}),
+        compiled_wf_1,
+    )
+
+    rydberg_drive = Fields(detuning=[detuning_term_0, detuning_term_1], rabi=[])
+    drives = {LevelCoupling.Rydberg: rydberg_drive}
+
+    expected_emulator_ir = EmulatorProgram(
+        register=geometry, duration=1.0, pulses=drives
+    )
+
+    emulator_ir = EmulatorProgramCodeGen(assignments=assignments).emit(program)
+
+    print(emulator_ir)
+    print(expected_emulator_ir)
+
+    assert emulator_ir == expected_emulator_ir
+
+
+def test_codegen_rabi_max_terms():
+    program = (
+        start.add_position((0, 0))
+        .add_position((0, 5))
+        .rydberg.rabi.amplitude.uniform.piecewise_linear(
+            [0.1, 0.8, 0.1], [-10, -10, 10, 10]
+        )
+        .var("mask_1")
+        .piecewise_linear([0.1, 0.8, 0.1], [-20, -20, 10, 10])
+        .location(0)
+        .scale(0.4)
+        .piecewise_linear([0.1, 0.8, 0.1], [-10, -10, 20, 20])
+        .parse_circuit()
+    )
+    mask_1_value = [0.5, 2.0]
+
+    mask_1 = RunTimeVector("mask_1")
+    location_0 = ScaledLocations({Location(0): cast(0.4)})
+    uniform_wf = program.sequence.pulses[rydberg].fields[rabi.amplitude].value[Uniform]
+    mask_1_wf = program.sequence.pulses[rydberg].fields[rabi.amplitude].value[mask_1]
+    location_0_wf = (
+        program.sequence.pulses[rydberg].fields[rabi.amplitude].value[location_0]
+    )
+
+    wf_0 = (1.0 * uniform_wf + mask_1_value[0] * mask_1_wf) + 0.4 * location_0_wf
+    wf_1 = 1.0 * uniform_wf + mask_1_value[1] * mask_1_wf
+
+    assignments = {"mask_1": mask_1_value}
+
+    compiled_wf_0 = CompiledWaveform(assignments, wf_0)
+    compiled_wf_1 = CompiledWaveform(assignments, wf_1)
+
+    geometry = Register(
+        TwoLevelAtom,
+        [(Decimal("0"), Decimal("0")), (Decimal("0"), Decimal("5"))],
+        Decimal("0"),
+    )
+
+    rabi_term_0 = RabiTerm(
+        RabiOperatorData(
+            RabiOperatorType.RabiSymmetric,
+            {0: Decimal("1")},
+        ),
+        compiled_wf_0,
+        None,
+    )
+
+    rabi_term_1 = RabiTerm(
+        RabiOperatorData(
+            RabiOperatorType.RabiSymmetric,
+            {1: Decimal("1")},
+        ),
+        compiled_wf_1,
+        None,
+    )
+
+    rydberg_drive = Fields(detuning=[], rabi=[rabi_term_0, rabi_term_1])
+
+    drives = {LevelCoupling.Rydberg: rydberg_drive}
+
+    expected_emulator_ir = EmulatorProgram(
+        register=geometry, duration=1.0, pulses=drives
+    )
+
+    emulator_ir = EmulatorProgramCodeGen(assignments=assignments).emit(program)
+
+    print(emulator_ir)
+    print(expected_emulator_ir)
```

Comment:
this is something forgot to delete?

Reply:
its in a test so I don't think this is an issue.

Reply:
its in the tests so I don't think this is an issue